### PR TITLE
upgrade nft

### PR DESCRIPTION
**Pull request by alyn509**
added an endpoint that accepts an NFT as payment, burns it, mints another one with a few attributes changed and send it back to the caller

### CODE REVIEW

File: nft-minter/src/nft_minting.rs [hunk 1]
```diff
@@ -126,6 +128,51 @@ pub trait NftMintingModule:
...
+
+        let sc_address = self.blockchain().get_sc_address();
+        let old_nft_data = self.blockchain().get_esdt_token_data(
+            &sc_address,
+            &old_nft.token_identifier,
+            old_nft.token_nonce,
+        );
+
+        require!(
+            old_nft_data.royalties > ROYALTIES_THRESHOLD,
```
Comment (CostinCarabas) [inline nft-minter/src/nft_minting.rs:150]: Why this `require`? You only allow upgrade for NFTs that have royalties higher than 10%?

Comment (CostinCarabas) [inline nft-minter/src/nft_minting.rs:150]: I think it should be like this: `old_nft_data.royalties > ROYALTIES_THRESHOLD,`

File: nft-minter/src/nft_minting.rs [hunk 2]
```diff
@@ -126,6 +128,51 @@ pub trait NftMintingModule:
...
+            "Unable to repair NFT"
```
Comment (CostinCarabas) [inline nft-minter/src/nft_minting.rs]: I think `repair` should be changed to `upgrade`

File: nft-minter/src/nft_minting.rs [hunk 3]
```diff
@@ -126,6 +128,51 @@ pub trait NftMintingModule:
...
+
+        self.send()
+            .esdt_local_burn(&old_nft.token_identifier, old_nft.token_nonce, &nft_amount);
+        let caller = self.blockchain().get_caller();
+
+        let nft_nonce = self.send().esdt_nft_create(
+            &old_nft.token_identifier,
+            &nft_amount,
+            &nft_name,
+            &BigUint::from(ROYALTIES_REPAIR),
```
Comment (CostinCarabas) [inline nft-minter/src/nft_minting.rs:162]: Why use different Royalties value?

File: nft-minter/tests/rust_test.rs [hunk 4]
```diff
@@ -393,6 +395,57 @@ fn buy_whitelist_test() {
...
+        &rust_biguint!(1),
+        &FIRST_ATTRIBUTES.to_vec(),
+        900,
+        None,
+        None,
+        None,
+        &uris_to_vec(FIRST_URIS),
+    );
+    nm_setup
+        .call_nft_upgrade(
```
Comment (CostinCarabas) [inline nft-minter/tests/rust_test.rs]: You should check the modified values.

**Review by CostinCarabas — CHANGES_REQUESTED** [review]

File: nft-minter/src/nft_minting.rs (hunk 1 above)
Comment (alyn509) [reply nft-minter/src/nft_minting.rs:150]: that was the request to mint again the NFT if the royalties are under a certain number

File: nft-minter/src/nft_minting.rs (hunk 3 above)
Comment (alyn509) [reply nft-minter/src/nft_minting.rs:162]: I sent you a link of the conversation in slack

File: nft-minter/tests/rust_test.rs (hunk 4 above)
Comment (alyn509) [reply nft-minter/tests/rust_test.rs]: the only element changing within this endpoint is the royalties attribute and it is changed by a constant value

**Review by CostinCarabas — DISMISSED** [review]

**Review by CostinCarabas — DISMISSED** [review]

**Pull request closed without merging**